[[ Bug 23104 ]] Fix calculation of player's native layer rect when stack is resized

livecodepanos opened the pull request:
This patch ensures the player's native layer rect is calculated correctly when the stack is resized.

livecodeian commented on engine/src/native-layer-mac.mm:
```diff
@@ -157,7 +157,16 @@
 void MCNativeLayerMac::doSetViewportGeometry(const MCRectangle &p_rect)
 {
-    doSetGeometry(m_object->getrect());
+    MCRectangle t_rect;
+    t_rect = m_object->getrect();
+    
+    if (m_object -> gettype() == CT_PLAYER)
+    {
+        MCPlayer *t_player = (MCPlayer *)m_object;
+        t_rect = t_player -> getvideorect(t_rect);
+    }
+    
+    doSetGeometry(t_rect);
```
@livecodepanos it's not a good idea to have code in MCNativeLayerMac that's specific to another control class. The bug arises from the player using a native layer that doesn't cover its entire rect. Perhaps we need a mechanism for getting the rect for a specific native layer from the control that it belongs to. Will need to give this some further thought.

livecodeian replied:
A simple fix may be to replace the original `doSetGeometry(m_object->getrect())` with `doSetGeometry(m_rect)`

livecodepanos committed:
[[ Bug 23104 ]] Fix calculation of player's native layer rect when stack is resized

This patch ensures the player's native layer rect is calculated correctly
when the stack is resized.

livecodeian commented:
@livecode-vulcan review ok 64d0f68

livecode-vulcan commented:
💙 review by @livecodeian ok 64d0f6878e30252bfe8160104c2621e099d18773

livecode-vulcan commented:
😎 test success 64d0f6878e30252bfe8160104c2621e099d18773

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/1019)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/427)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/426)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/425)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/426)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2544)
* try-community-universal-ios-iphoneos14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.5/builds/22)
* try-community-universal-ios-iphoneos14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.4/builds/73)
* try-community-universal-ios-iphonesimulator14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.5/builds/22)
* try-community-universal-ios-iphonesimulator14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.4/builds/73)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2571)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2560)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2558)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2595)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2531)
